app/controllers/coronavirus_form/are_you_a_manufacturer_controller.rb

```diff
@@ -1,37 +1,35 @@
 # frozen_string_literal: true
 
 class CoronavirusForm::AreYouAManufacturerController < ApplicationController
-  before_action :check_first_question_answered, only: :show
+  before_action :check_first_question_answered, except: :submit
 
   def show
     session["are_you_a_manufacturer"] ||= []
-    render "coronavirus_form/#{PAGE}"
+    super
```

Comment:
I think if you use `session.dig("are_you_a_manufacturer") you might not need to memoize it here, and then could remove the method.